### PR TITLE
used botex to add mailto link in the website footer

### PR DESCRIPTION
**Pull request by frozenmafia**

### CODE REVIEW

File: docusaurus.config.js [hunk 1]
```diff
@@ -162,6 +165,10 @@ const config = {
                 label: 'YouTube',
                 href: youtubeUrl,
               },
+              {
+                label:'Email',
+                href:`mailto:${unscramble(obfuscatedEmail, key)}`
```
Comment (sergefdrv) [inline docusaurus.config.js]: I'm afraid, it's not that simple. The call to `unscramble` and setting `href` should happen dynamically in an event handler attached to the element, e.g. `mouseover`. As it is, the email address will still appear in plain text in the generated website.

File: docusaurus.config.js [hunk 2]
```diff
@@ -17,7 +18,9 @@ const siteLicense = {
   icon: "https://i.creativecommons.org/l/by/4.0/80x15.png",
   url: "http://creativecommons.org/licenses/by/4.0/",
 };
-
+const email = 'contact@replica-io.dev';
+const key = "abc123";
```
Comment (sergefdrv) [inline docusaurus.config.js]: Let's use a more random key, e.g. generated with `pwgen -N1`.

Comment (sergefdrv) [reply docusaurus.config.js]: And maybe pick a more specific name for the variable, e.g. `emailSalt`.

File: docusaurus.config.js [hunk 3]
```diff
@@ -162,6 +165,10 @@ const config = {
                 label: 'YouTube',
                 href: youtubeUrl,
               },
+              {
+                label:'Email',
```
Comment (sergefdrv) [inline docusaurus.config.js:169]:
```suggestion
                label:'Contact Email',
```

**Review by sergefdrv — CHANGES_REQUESTED** [review]
I suggest using the following commit message subject: "Add contact link to website footer".  You can mention `botex` in the commit message body (after a separating empty line).

The `Signed-off-by` line is used to certify the copyright status of code contributions. Therefore, it should include your real name (sorry, no pseudonyms or anonymous contributions).

**Comment by sergefdrv** [comment]
> I suggest using the following commit message subject: "Add contact link to website footer". You can mention `botex` in the commit message body (after a separating empty line).

In general, I think [this](https://cbea.ms/git-commit/) is a good article on how to write good commit messages.

File: docusaurus.config.js [hunk 4]
```diff
@@ -1,76 +1,77 @@
 // @ts-check
 // Note: type annotations allow type checking and IDEs autocompletion
 
-import { themes as prismThemes } from 'prism-react-renderer';
+import { themes as prismThemes } from "prism-react-renderer";
```
Comment (sergefdrv) [inline docusaurus.config.js:4]: If you'd like to change coding style, let's do that in a separate commit.

File: docusaurus.config.js [hunk 5]
```diff
@@ -1,76 +1,77 @@
...
+const siteGitHubUrl = "https://github.com/replica-io/replica-io.github.io";
 const siteLicense = {
   name: "CC-BY-4.0",
   icon: "https://i.creativecommons.org/l/by/4.0/80x15.png",
   url: "http://creativecommons.org/licenses/by/4.0/",
 };
-const email = 'contact@replica-io.dev';
+const emailSalt = "contact@replica-io.dev";
 const key = "abc123";
-const obfuscatedEmail = scramble(email, key);
```
Comment (sergefdrv) [inline docusaurus.config.js]: I encoded the email using `botex` CLI. This should be something like:
```
const contactEmailKey = "CHKXRg4cdMZ0XADF";
const obfuscatedContactEmail = "BwC1311181fB4C1sBzmC14162k1dByC112z16132t1o1jBh";
```

File: docusaurus.config.js [hunk 6]
```diff
@@ -81,94 +82,108 @@ const config = {
...
+                      <div>
+                        <a id="emailLink" href="#" target="_blank" rel="noopener noreferrer" class="footer__link-item" onclick="unscrambleEmail(event)">
+                          Email
+                          <svg width="13.5" height="13.5" aria-hidden="true" viewBox="0 0 24 24" class="iconExternalLink">
+                            <path fill="currentColor" d="M21 13v10h-21v-19h12v2h-10v15h17v-8h2zm3-12h-10.988l4.035 4-6.977 7.07 2.828 2.828 6.977-7.07 4.125 4.172v-11z"></path>
+                          </svg>
+                        </a>
+                      </div>
+                    `,
+              },
```
Comment (sergefdrv) [inline docusaurus.config.js]: Let's try setting `href: "#contact-email"` to all contact email links (footer and later navigation bar) instead of using raw HTML. Then we could select those elements with `document.querySelectorAll('a[href="#contact-email"]')` and attach the `onclick` event handler to them in a script or [client module](https://docusaurus.io/docs/api/docusaurus-config#clientModules).

**Review by sergefdrv — CHANGES_REQUESTED** [review]

File: docusaurus.config.js (hunk 4 above)
Comment (frozenmafia) [reply docusaurus.config.js:4]: No, I did it by mistake. Let it remain the way it is.

File: docusaurus.config.js [hunk 7]
```diff
@@ -78,99 +80,109 @@ const config = {
...
                 href: youtubeUrl,
               },
             ],
           },
+          {
+            title: "Contact Us",
+            items: [
+              {
+                label: "Email",
+                href: emailSalt,
```
Comment (sergefdrv) [inline docusaurus.config.js]:
```suggestion
                href: "#contact-email",
```

File: docusaurus.config.js [hunk 8]
```diff
@@ -1,73 +1,75 @@
...
+const youtubeUrl = "https://www.youtube.com/@Replica_IO";
+const siteGitHubUrl = "https://github.com/replica-io/replica-io.github.io";
 const siteLicense = {
   name: "CC-BY-4.0",
   icon: "https://i.creativecommons.org/l/by/4.0/80x15.png",
   url: "http://creativecommons.org/licenses/by/4.0/",
 };
-
-
+const emailSalt = "#contact-email";
```
Comment (sergefdrv) [inline docusaurus.config.js]:
```suggestion
```

File: static/js/unscrambleEmail.js [hunk 9]
```diff
@@ -0,0 +1,17 @@
+import { unscramble } from "botex";
+
+export function onRouteDidUpdate({ location, previousLocation }) {
+  if (typeof window !== 'undefined') {
+    setTimeout(() => {
```
Comment (sergefdrv) [inline static/js/unscrambleEmail.js]: Do we really need a timeout here? Docusaurus docs [say](https://docusaurus.io/docs/advanced/client#client-module-lifecycles): 

> If you need to manipulate the new page's DOM, you'll likely want to use `onRouteDidUpdate`, which will be fired as soon as the DOM on the new page **has mounted**.

File: static/js/unscrambleEmail.js [hunk 10]
```diff
@@ -0,0 +1,21 @@
...
+
+export function onRouteDidUpdate({ location, previousLocation }) {
+  if (typeof window !== "undefined") {
+    setTimeout(() => {
+      document.body.addEventListener("click", function (event) {
+        if (event.target.matches('a[href="/#contact-email"]')) {
+          event.preventDefault();
+          const contactEmailKey = "CHKXRg4cdMZ0XADF";
+          const obfuscatedContactEmail =
+            "BwC1311181fB4C1sBzmC14162k1dByC112z16132t1o1jBh";
```
Comment (sergefdrv) [inline static/js/unscrambleEmail.js]: Let's move these lines out of the function body. Since they are local in this file, we can use shorter names like `key` (or `botexKey`) and `obfuscatedEmail`.

File: static/js/unscrambleEmail.js [hunk 11]
```diff
@@ -0,0 +1,21 @@
+import { unscramble } from "botex";
+
+export function onRouteDidUpdate({ location, previousLocation }) {
+  if (typeof window !== "undefined") {
```
Comment (sergefdrv) [inline static/js/unscrambleEmail.js]:
```suggestion
  if (ExecutionEnvironment.canUseDOM) {
```

You'd also need to add `import ExecutionEnvironment from '@docusaurus/ExecutionEnvironment';`.

**Review by sergefdrv — CHANGES_REQUESTED** [review]

File: static/js/unscrambleEmail.js (hunk 9 above)
Comment (frozenmafia) [reply static/js/unscrambleEmail.js]: Yes , without timeout it was not working

Comment (sergefdrv) [reply static/js/unscrambleEmail.js]: I experimented a bit, and the following seems to work well (provided that email links refer to `/#contact-email` instead of `#contact-email`):

```js
import ExecutionEnvironment from '@docusaurus/ExecutionEnvironment';
import { unscramble } from "botex";

const botexKey = "CHKXRg4cdMZ0XADF";
const obfuscatedEmail = "BwC1311181fB4C1sBzmC14162k1dByC112z16132t1o1jBh";

export function onRouteUpdate({ location, previousLocation }) {
  if (ExecutionEnvironment.canUseDOM) {
    if (location.pathname === '/' && location.hash === '#contact-email') {
      const email = unscramble(obfuscatedEmail, botexKey);
      window.location.href = `mailto:${email}`;
    }
  }
}
```
